load/config.go

```diff
@@ -12,16 +13,49 @@ const (
 	defaultKVMaxValueSize = 4096
 )
 
+type Target int
+
+const (
+	TargetKV Target = iota
+	TargetPeering
+)
+
+func (t Target) GoString() string { return t.String() }
+
+func (t Target) String() string {
+	switch t {
+	case TargetKV:
+		return "kv"
+	case TargetPeering:
+		return "peering"
+	default:
+		return ""
+	}
+}
+
 type Config struct {
 	KV      *KVConfig
 	Peering *PeeringConfig
+	Target  Target
 }
 
 func (c *Config) Normalize() error {
+	if c.KV != nil && c.Peering != nil {
+		return errors.New("error validating configuration, cannot configure both KV and Peering, must choose one or the other")
+	}
+
```

Comment:
are we going to allow both KV and peering to be not-nil? To make it consistent with the following two `if`, maybe we can return an error saying `target must be one of [kv, peering, ....]`

Reply:
Yeah, the intent is for the `kv` and `peering` configs to be mutually exclusive. I think this is the case, right? I have a test for it below too named `"config.KV and config.Peering are mutually exclusive"`.

I think the message here is the same as what you are suggesting. Is this what you were suggesting or am I missing something?

Reply:
Oh, I misunderstood the condition. It looks good.